tests/admin/tournaments/updateTournamentsPosition.test.ts

```diff
@@ -0,0 +1,124 @@
+      position: number;
+    }[];
+  };
+
+  after(async () => {
+    await database.user.deleteMany();
+  });
+
+  before(async () => {
+    admin = await createFakeUser({ type: UserType.orga, permissions: [Permission.admin] });
+    nonAdminUser = await createFakeUser();
+    adminToken = generateToken(admin);
+
+    tournaments = await fetchTournaments();
+
+    validBody = {
+      tournaments: [
+        {
+          id: tournaments[0].id,
+          position: 1,
+        },
+        {
+          id: tournaments[1].id,
+          position: 0,
+        },
+      ],
+    };
+  });
+
+  it('should error as the user is not authenticated', () =>
+    request(app).patch(`/admin/tournaments`).send(validBody).expect(401, { error: Error.Unauthenticated }));
+
+  it('should error as the user is not an administrator', () => {
+    const userToken = generateToken(nonAdminUser);
+    return request(app)
+      .patch(`/admin/tournaments`)
+      .send(validBody)
+      .set('Authorization', `Bearer ${userToken}`)
+      .expect(403, { error: Error.NoPermission });
+  });
+
+  it('should fail with an internal server error', async () => {
+    sandbox.stub(tournamentOperations, 'updateTournamentsPosition').throws('Unexpected error');
+
+    await request(app)
+      .patch(`/admin/tournaments`)
+      .send(validBody)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .expect(500, { error: Error.InternalServerError });
+  });
+
+  it("should throw an error as tournament's id does not exist", async () => {
+    await request(app)
+      .patch(`/admin/tournaments`)
+      .send({
+        tournaments: [
+          {
+            id: 'aaaaaa',
+            position: 1,
+          },
+          {
+            id: 'aaaaaa',
+            position: 0,
+          },
+        ],
+      })
+      .set('Authorization', `Bearer ${adminToken}`)
+      .expect(404, { error: Error.NotFound });
+  });
+
+  it("should throw an error as tournament's position is not a number", async () => {
+    await request(app)
+      .patch(`/admin/tournaments`)
+      .send({
+        tournaments: [
+          {
+            id: tournaments[0].id,
+            position: 'aaaaaa',
+          },
+          {
+            id: tournaments[1].id,
+            position: 0,
+          },
+        ],
+      })
+      .set('Authorization', `Bearer ${adminToken}`)
+      .expect(400);
+  });
+
+  it('should successfully update the tournaments', async () => {
+    const result = await request(app)
+      .patch(`/admin/tournaments`)
+      .send(validBody)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .expect(200);
+
+    expect(
+      result.body.find((a: { id: string; position: number }) => a.id === validBody.tournaments[0].id).position,
+    ).to.equal(validBody.tournaments[0].position);
+    expect(
+      result.body.find((a: { id: string; position: number }) => a.id === validBody.tournaments[1].id).position,
+    ).to.equal(validBody.tournaments[1].position);
+  });
```

Comment:
pareil, on veut une erreur si on envoie qu'un seul tournoi ?

Reply:
C'est le test là donc comme j'ai test avec 2 values bah je reste avec 2 values ici mais ca marche bien pour une aussi comme on a juste un for